cmd/root/bundle.go

```diff
@@ -81,6 +81,19 @@ func configureBundle(cmd *cobra.Command, b *bundle.Bundle) (*bundle.Bundle, diag
 
 	// Configure the workspace profile if the flag has been set.
 	diags = diags.Extend(configureProfile(cmd, b))
+	if diags.HasError() {
+		return b, diags
+	}
+
+	// Set the auth configuration in the command context. This can be used
+	// downstream to initialize a API client.
+	client, err := b.WorkspaceClientE()
```

Comment:
Initializing a client performs auth resolution.

If we perform this here then stuff that must run as fast as possible may take longer after this change.

I'm thinking about e.g. command completion.

It would be good to figure out the guarantees and include a comment on what we assume the guarantee to be.

Reply:
Discussed offline on slack, initializing the workspace client is expected to be fast so this is fine. 

Added some context here though.

Also I would not expect PreRunE to be called in the completions command (not verified). Looking at the code it seems the completions functionality is provided as a completely independent command in the binary.